rust/worker/src/segment/metadata_segment.rs

```
@@ -418,7 +481,210 @@ impl<'a> SegmentWriter<'a> for MetadataSegmentWriter {
+                                                    );
+                                                }
+                                                None => {}
+                                            }
+                                        }
+                                        MetadataValue::Float(value) => {
+                                            match &self.f32_metadata_index_writer {
+                                                Some(writer) => {
+                                                    let _ = writer.delete(
+                                                        key,
+                                                        *value as f32,
+                                                        segment_offset_id,
+                                                    );
+                                                }
+                                                None => {}
+                                            }
+                                        }
+                                        MetadataValue::Int(value) => {
+                                            match &self.u32_metadata_index_writer {
+                                                Some(writer) => {
+                                                    let _ = writer.delete(
+                                                        key,
+                                                        *value as u32,
+                                                        segment_offset_id,
+                                                    );
+                                                }
+                                                None => {}
+                                            }
+                                        }
+                                    }
+                                }
+                            }
+                            None => {}
```

nit: This should return an error since it violates invariance that record to be deleted wasn't found in segment storage

---

Hammad and I have talked about this in the past. We have to decide on a plan of action for such cases. Possibilities:
1. Ignore only such records. This could have a bearing on the correctness for e.g. if future records are causally related to this operation thus ignoring only this record and not ignoring the future ones. Although in practice, I am not sure when can such a causal dependency happen.
2. Trim the entire suffix starting from this log offset.

For now, we propagate these errors and later we'll decide on a plan of action for these based on error types.